src/Data/Nat/Primality.agda

```diff
@@ -127,9 +339,11 @@ euclidsLemma m n {p@(suc (suc _))} p-prime p∣m*n = result
 
   -- if the GCD of m and p is greater than one, then it must be p and hence p ∣ m.
   ... | Bézout.result d@(suc (suc _)) g _ with d ≟ p
-  ...   | yes refl = inj₁ (GCD.gcd∣m g)
-  ...   | no  d≢p  = contradiction (GCD.gcd∣n g) (p-prime 2≤d d<p)
-    where 2≤d = s≤s (s≤s z≤n); d<p = flip ≤∧≢⇒< d≢p (∣⇒≤ (GCD.gcd∣n g))
+  ...   | yes d≡p@refl = inj₁ (GCD.gcd∣m g)
+  ...   | no  d≢p = contradiction d∣p λ d∣p → pr (boundedComposite≢ d≢p d∣p)
```

Comment:
It's how it was proved before... which I simply cleaned up a bit. But I agree, this style is annoying!

Reply:
NB the 'introduction' is a deliberate pun: we need a proof of `¬ (d ∣ p)` for the contradiction, so we introduce a hypothesis of the same name, and then use `pr (boundedComposite≢ d≢p d∣p)` to yield a proof of `⊥`. Of course, the proof could directly appeal to `⊥-elim` or `with () ← ...` but elsewhere in the history of v2.0, you have emphasised using `contradiction` over the latter form, and I find the `⊥-elim` proofs introduce redundant (syntactic) `import`s. Matter of taste, or one for the style guide to rule (less or more dogmatically) upon?

Reply:
I think it can be shuffled around a bit to be simpler. Something like `contradiction (boundedComposite≢ d≢p d∣p) pr`. I think it was written like this as a consequence of how I was thinking about this bit of the proof - "I know this case is impossible, because d divides p but if d isn't p then d can't divide p". Clean-up there is very welcome

Reply:
Double-plus :+1:!